rocks-bin/src/rockspec/write_rockspec.rs

```diff
@@ -118,132 +121,151 @@ fn parse_version(input: String) -> Result<LuaDependency> {
     LuaDependency::from_str(format!("lua {}", input).as_str())
 }
 
-pub async fn write_rockspec(data: WriteRockspec) -> Result<()> {
-    let mut spinner = Spinner::new(Spinners::Dots, "Fetching repository metadata...".into());
-
-    // [data.labels, data.license].into_iter().all(Option::is_some);
-
-    let repo_metadata = github_metadata::get_metadata_for(None)
-        .await?
-        .unwrap_or_else(|| RepoMetadata {
-            name: std::env::current_dir()
-                .expect("unable to get current working directory")
-                .file_name()
-                .unwrap_or_default()
-                .to_string_lossy()
-                .to_string(),
-            description: None,
-            license: None,
-            contributors: vec![users::get_current_username()
-                .unwrap_or_default()
-                .to_string_lossy()
-                .to_string()],
-            labels: None,
-        });
-
-    spinner.stop();
-
-    let mut editor = rustyline::Editor::<(), _>::new()?;
-
-    // TODO: Make prompts coloured
-
-    // let mut stdout = BufferedStandardStream::stdout(ColorChoice::Always);
-    // stdout.set_color(ColorSpec::new().set_fg(Some(Color::Green)))?;
-
-    let package_name = data.name.map(Ok).unwrap_or_else(|| {
-        parse!(
-            editor
-                .readline(format!("Package Name (empty for '{}'): ", repo_metadata.name).as_str()),
-            identity,
-            repo_metadata.name
-        )
-    })?;
-
-    let description = data.description.map(Ok).unwrap_or_else(|| {
-        parse!(
-            editor.readline(
-                format!(
-                    "Description (empty for '{}'): ",
-                    repo_metadata
-                        .description
-                        .as_ref()
-                        .unwrap_or(&"*** enter a description ***".to_string())
+pub async fn write_rockspec(cli_flags: WriteRockspec) -> Result<()> {
+    let (package_name, description, labels, license, lua_versions, maintainer) = match (
+        cli_flags.name,
+        cli_flags.description,
+        cli_flags.labels,
+        cli_flags.license,
+        cli_flags.lua_versions,
+        cli_flags.maintainer,
+    ) {
+        ( // If all parameters are provided then don't bother prompting the user
+            Some(name),
+            Some(description),
+            Some(labels),
+            Some(license),
+            Some(lua_versions),
+            Some(maintainer),
+        ) => Ok::<_, eyre::Report>((name, description, labels, license, lua_versions, maintainer)),
```

Comment:
suggestion: I think it's better to destructure the struct, rather than match on all fields individually, because the compiler will let us know if we add a field and forget about it in our pattern match (ditto for the other match arm).

```suggestion
    let (package_name, description, labels, license, lua_versions, maintainer) = match (cli_flags) {
        WriteRockspec {
            // If all parameters are provided then don't bother prompting the user
            name: Some(name),
            description: Some(description),
            labels: Some(labels),
            license: Some(license),
            lua_versions: Some(lua_versions),
            maintainer: Some(maintainer),
            directory: _,
        } => Ok::<_, eyre::Report>((name, description, labels, license, lua_versions, maintainer)),

```

Reply:
Oh, I totally forgot you could do that. Good catch!